### PR TITLE
[IMP] stock, purchase, sale: inventory back to basics

### PR DESCRIPTION
**Pull request by AhmedAmirK**
Applies various small changes:

- [IMP] stock, stock_account: add dates to stock move name (inventory adjustments)
- [IMP] delivery,stock: add pack type weight
- [IMP] {purchase, sale}_stock: add Incoterm Location field
- [IMP] stock_picking_batch: default show Batch Transfer
- [IMP] purchase: add supplier info with sudo qty 1
- [IMP] sale_purchase: add related PO warning
- [FIX] product: order product labels by product
- [IMP] product: default filter vendor pricelist w/ active products
- [IMP] stock: relax picking locations readonly states
- [IMP] stock: add consumables replenish button 
- [IMP] stock_picking_batch: pick type auto batch not active by default
- [IMP] stock: add stock quant import template
- [FIX] sale,stock,mrp,purchase: add uom precision

Task Id: 2764771
COM PR: odoo/odoo#91640

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by ticodoo** [comment]
Hello @odoo/rd-accounting we need your help with a failing test related to the commit "[IMP] purchase: Supplier info now added with sudo with qty 1" in this PR.

**TLDR version:** Can 2 res.partner.bank records be assigned to the same res.partner and have different `company_id` values? If not, do you think we can delete the test `TestPurchaseToInvoice.test_multicompany_partner_bank`? If yes, do you see a way the test can be adapted to work correctly?

The test: https://github.com/odoo/odoo/blob/814659b73dd9a9730482167a24facd4ec3b7e35d/addons/purchase/tests/test_purchase_invoice.py#L331-L397
is currently failing even though the change shouldn't effect it (spoiler: there's something really strange going on with the caching in the test). While investigating I noticed that the test was written for 14.0 when this was the case (orig PR: https://github.com/odoo/odoo/pull/84305)
https://github.com/odoo/odoo/blob/843e57694dd73ed4b7fca0460a4cf9b166820843/odoo/addons/base/models/res_bank.py#L86
I.e. each bank has its own company_id
But you guys changed this in saas-15.2 (PR: https://github.com/odoo/odoo/pull/81362) to:
https://github.com/odoo/odoo/blob/73ebd913b776a652d92a46f5e4e87259718c7432/odoo/addons/base/models/res_bank.py#L86
So something weird is going on with the cache after these assignments:
https://github.com/odoo/odoo/blob/814659b73dd9a9730482167a24facd4ec3b7e35d/addons/purchase/tests/test_purchase_invoice.py#L354-L355
Because the `partner.company_id` stays unassigned afterwards and the test fails because the `partner.bank_ids` unlinks from `partner_bank_b` sometime around
https://github.com/odoo/odoo/blob/814659b73dd9a9730482167a24facd4ec3b7e35d/addons/purchase/tests/test_purchase_invoice.py#L390-L391
Note that if you step through the test in the debugger, it doesn't do the unlinking when you step into the first `po_a.with_company(company_a).action_create_invoice()`

Due to the saas-15.2 change, I think the test is no longer needed. If you guys disagree then help would be appreciated for how to adapt the test so it works correctly.

**Review by Feyensv — COMMENTED** [review]
Quick check, mostly "linting" comments ;)

File: addons/sale_stock/report/sale_order_report_templates.xml [hunk 1]
```diff
@@ -4,7 +4,12 @@
         <xpath expr="//div[@name='expiration_date']" position="after">
             <div class="col-auto col-3 mw-100 mb-2" t-if="doc.incoterm" groups="sale_stock.group_display_incoterm">
                 <strong>Incoterm:</strong>
-                <p class="m-0" t-field="doc.incoterm.code"/>
+                <t t-if="doc.incoterm_location">
+                    <p t-esc="'{} ({})'.format(doc.incoterm.code, doc.incoterm_location)" class="m-0"/>
+                </t>
+                <t t-else="">
+                    <p t-field="doc.incoterm.code" class="m-0"/>
+                </t>
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/sale_stock/report/sale_order_report_templates.xml]: Note that the white spaces change between the versions, it might have importance in some applications (most likely not here though)

**Review by ticodoo — COMMENTED** [review]
Overall looks good, mainly little cleanup things to do. I will review the pack type weight commit after its related changes have been merged into the same commit.

Only other small thing is for your commit messages. They should all have `Task: 2764771` added to them (i.e. this is the task id number) so people know where to look for the original spec. Also, not a big deal for this task since its a back2basics + your first, but the first line of your commit message should also ideally fit within the 73 characters or less length for readability and should say what the commit is doing,

e.g.: instead of `[IMP] purchase_stock: Incoterm Location field in PO and PO reports`
```
[IMP] {purchase, sale}_stock: add Incoterm Location field

This commit adds a new Incoterm Location field that will be printed
in PO and SO reports if filled in.

Part 10 of Task: 2764771
```

And lastly, since this PR will be squashed by the runbot and the PR title and body will be used for the final merge commit, can you update the PR title and description to match the commit standard? Something like: 

```
[IMP] purchase, stock: apply inventory back2basics

Applies many small back2basics changes, see individual commits for
additional details:
- replace forecast label with info circle
- add schedule/accounting date to stock move name
- .... etc
```

**Review by svs-odoo — APPROVED** [review]
Hello,
Here some comments. It's mainly suggestions for shorter code or formating purpose, everything else looks good to me :+1:

**Review by ticodoo — COMMENTED** [review]
So close! Just one last fix

File: addons/stock/models/stock_quant.py [hunk 2]
```diff
@@ -877,6 +877,10 @@ def _get_inventory_move_values(self, qty, location_id, location_dest_id, out=Fal
             name = _('Product Quantity Confirmed')
         else:
             name = _('Product Quantity Updated')
+
+        if self.inventory_date:
+            name += f' [Scheduled on {self.inventory_date}]'
```
Comment (ticodoo) [inline addons/stock/models/stock_quant.py:890]: Ah sorry, I missed this earlier. You need to add a `_()` around this for the translation. Also its preferred to use %s over f'  (when only one variable to replace in a string): https://www.odoo.com/documentation/15.0/developer/misc/other/guidelines.html#use-translation-method-correctly

Same thing applies to  ```res_move['name'] += f' [Accounted on {force_period_date}]'``` in the stock_account module

File: addons/delivery/models/stock_picking.py [hunk 3]
```diff
@@ -25,6 +26,8 @@ def _compute_weight(self):
                     weight += ml.product_uom_id._compute_quantity(
                         ml.qty_done, ml.product_id.uom_id) * ml.product_id.weight
             else:
+                if package.package_type_id:
+                    weight = package.package_type_id.base_weight
```
Comment (ticodoo) [inline addons/delivery/models/stock_picking.py:30]: you can delete this now that you have the value being set outside of the if/else statement

**Review by ticodoo — COMMENTED** [review]
Sorry, final check caught a couple small things to change :crying_cat_face:

**Comment by ticodoo** [comment]
Thanks!
@robodoo rebase-merge r+

**Comment by william-andre** [comment]
Rebase-merge: the mesage will be
> Applies various small changes check individual commits and task for details

But there is only one commit :thinking:

**Comment by ticodoo** [comment]
@robodoo r-

@william-andre you caught me, I was being lazy and hoped people would just look at the PR since the commits will be preserved. I'll update it properly

**Comment by ticodoo** [comment]
@robodoo r+
